src/garage/torch/algos/vpg.py

```diff
@@ -131,54 +131,56 @@ def train_once(self, itr, paths):
                 * average_return: (float)
 
         """
-        valids, obs, actions, rewards = self.process_samples(itr, paths)
+        obs, actions, rewards, valids, baselines = self.process_samples(
+            itr, paths)
 
-        loss = self._compute_loss(itr, paths, valids, obs, actions, rewards)
+        loss = self._compute_loss(itr, obs, actions, rewards, valids,
+                                  baselines)
 
-        # Memorize the policy state_dict
         self._old_policy.load_state_dict(self.policy.state_dict())
 
         self._optimizer.zero_grad()
         loss.backward()
 
         kl_before = self._compute_kl_constraint(obs).detach()
-        self._optimize(itr, paths, valids, obs, actions, rewards)
+        self._optimize(itr, obs, actions, rewards, valids, baselines)
 
         with torch.no_grad():
-            loss_after = self._compute_loss(itr, paths, valids, obs, actions,
-                                            rewards)
+            loss_after = self._compute_loss(itr, obs, actions, rewards, valids,
+                                            baselines)
             kl = self._compute_kl_constraint(obs)
             policy_entropy = self._compute_policy_entropy(obs)
-            average_return = self._log(itr, paths, loss.item(),
-                                       loss_after.item(), kl_before.item(),
-                                       kl.item(),
-                                       policy_entropy.mean().item())
+            average_return = self.evaluate_performance(
+                itr, {
+                    'paths': paths,
+                    'loss_before': loss.item(),
+                    'loss_after': loss_after.item(),
+                    'kl_before': kl_before.item(),
+                    'kl': kl.item(),
+                    'policy_entropy': policy_entropy.mean().item()
+                })
 
         self.baseline.fit(paths)
```

Comment:
Shall we fit baseline with the augmented reward? Are you using the original reward or augmented reward here?

Reply:
I recommend you fit with augmented reward -- you want the baseline to be as accurate as possible. This only works if the baseline input has enough information to predict the augmented reward, though.

Reply:
Here, the original reward is fed to the baseline. This is inherited from the old implementation of torch/VPG. If we were to fix this I think we better make it in a separate PR.